app/service/update-appeal-service.ts

```diff
@@ -162,11 +194,42 @@ export default class UpdateAppealService {
       hearingRequirements: {},
       respondentDocuments: respondentDocuments,
       documentMap: documentMap,
-      directions: directions
+      directions: directions,
+      timeExtensionEventsMap: timeExtensionEventsMap,
+      timeExtensions: timeExtensions
     };
 
     req.session.appeal.askForMoreTime = {};
-    req.session.appeal.previousAskForMoreTime = caseData.timeExtensions || [];
+    req.session.appeal.previousAskForMoreTime = [];
+
+    if (caseData.timeExtensions) {
+      caseData.timeExtensions.forEach(timeExtension => {
+        const askForMoreTime = this.convertToAskForMoreTime(timeExtension.value, documentMap);
+        if (timeExtension.value.status === 'inProgress' && timeExtension.value.state === ccdCase.state) {
+          req.session.appeal.askForMoreTime = askForMoreTime;
+        } else {
+          req.session.appeal.previousAskForMoreTime.push(askForMoreTime);
+        }
+      });
+    }
+  }
```

Comment:
I think something has gone wrong with the merge here as I don't think this is now the logic on master. We don't load askForMoreTime from CCD anymore as there is no save for later. previousAskForMoreTime was not converted to a frontend object as we were losing fields that were filled in when the TCW reviewed the time extensions. I guess you need to see parse them though to view them on the frontend.